wp/lib/bap_wp/src/compare.ml

```diff
@@ -314,3 +332,16 @@ let compare_subs_mem_eq : comparator * comparator =
     Constr.mk_clause [] (mem_eq :: pre_eqs), env1, env2
   in
   postcond, hyps
+
+let compare_subs_mem_init : comparator * comparator =
+  let postcond ~original ~modified ~rename_set:_ =
+    let ctx = Env.get_context original.env in
+    let post = Constr.trivial ctx in
+    post, original.env, modified.env
+  in
+  let hyps ~original ~modified ~rename_set:_ =
+    let pre, env1 = Pre.init_mem original.env original.mem in
```

Comment:
Can you explain why we only need to initialize the mem for the original binary?

Reply:
Well, I'm thinking this will mostly apply when the two memories are equal. Maybe I can explicitly introduce a dependency between the flags.

Suggestions are welcome.

Reply:
So if the two binaries have the same memories, but at different offsets, then this hypothesis would not apply, right?

Reply:
Indeed. We need to be careful here: if we initialize `mem_mod`, and it's different than `mem_orig` and then assume that they are equal, we'll get an inconsistent assumption, and get spurious `UNSAT`s.

I decided to leave out all assumptions involving `mem_mod` for starters.